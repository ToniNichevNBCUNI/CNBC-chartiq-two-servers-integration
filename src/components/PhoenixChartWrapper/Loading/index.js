import React from 'react';

const styles = require('./styles.scss');

/**
 * HELPER COMPONENT TO DISPLAY LOADING ... SCREEN AND HANDLE ERRORS
 * @param {} props 
 */
const Loading = (props) => {
  if (props.error) {
    return (<div className={styles.wrapper}>
              <h3>Error loading ChartIQ component!</h3> 
              <p><b>{props.error.message}</b></p>
              <p>{props.error.stack}</p>
              <div><button onClick={ props.retry }>Retry</button></div>
            </div>);
  } else {
    return (
    <div className={styles.wrapper}>
      <div className={styles.timeRange}><p> ... </p></div>
      <h1>Loading . . . </h1>
    </div>)
  }
} 

export default Loading;