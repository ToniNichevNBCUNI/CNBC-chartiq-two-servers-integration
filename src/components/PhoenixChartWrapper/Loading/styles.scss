.errorWrapper {
  background:none;
  border: 1px solid silver;
  background: red;
  color: white;
  height: 352px;

  .msg {
    position: relative;
    top: 35%;
    text-align: center;

    span {
      color: #000;
      margin: 0;
      padding: 2px;
      border-radius: 3px;
      line-height: inherit;
      font-size: 12px;
      font-family: inherit;
      background: white;
    }
  }
}

.wrapper {
  background:none;
  border: 1px solid silver;
  padding-bottom: 10px;
  height: 440px;
  
  h1 {
    position: relative;
    top: 35%;
    text-align: center;
    color: #636363;
  }

  .timeRange {
    text-align: center;
    position: relative;
    height: 44px;
    border-bottom: 1px solid silver;

    p {
      margin: 0;
    }
  }
  
  h3 {
    background: red;
    text-align: center;
  }
  div{
    text-align: center;
  }
}