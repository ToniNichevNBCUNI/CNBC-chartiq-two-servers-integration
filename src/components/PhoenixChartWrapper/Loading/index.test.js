import Loading from '.';

describe('Loading component', () => {
  it('should render default loading component', () => {
    const tree = getSnapshot(<Loading />);
    expect(tree).toMatchSnapshot();
  });

  it('should render night mode loading component', () => {
    const queryParams = {
      theme: 'night'
    };
    const tree = getSnapshot(<Loading queryParams={queryParams} />);
    expect(tree).toMatchSnapshot();
  });
});
