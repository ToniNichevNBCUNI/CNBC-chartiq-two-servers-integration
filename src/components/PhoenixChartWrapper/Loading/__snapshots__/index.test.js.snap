// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render default loading component 1`] = `
<div
  className="wrapper"
>
  <div
    className="timeRange"
  >
    <p>
       ... 
    </p>
  </div>
  <div>
    Loading . . . 
  </div>
</div>
`;

exports[`Loading component should render night mode loading component 1`] = `
<div
  className="nightModeWrapper"
>
  <div
    className="timeRange"
  >
    <p>
       ... 
    </p>
  </div>
  <div>
    Loading . . . 
  </div>
</div>
`;
